numpy/__init__.pyi

```diff
@@ -463,27 +465,24 @@ class float32(floating):
 class float64(floating):
     def __init__(self, __value: Optional[SupportsFloat] = ...) -> None: ...
 
-class complexfloating(inexact): ...  # type: ignore
+class complexfloating(inexact, Generic[_FloatType]):  # type: ignore
+    @property
+    def real(self) -> _FloatType: ...
+    @property
+    def imag(self) -> _FloatType: ...
```

Comment:
Aren't these properties of `np.number` anyway?

Reply:
They are and they were already defined for `np.number` some time ago:

https://github.com/numpy/numpy/blob/32b3f82e4862d85c5aebba2b2b82a931038cd103/numpy/__init__.pyi#L381-L387

What makes `np.complexfloating` somewhat unique is that the return type of 
`.real` and `.imag` is not `type(self)`; it's one of the `np.floating` subclasses.

Reply:
I think we forgot that integers support real and imag too, to match python

Reply:
> I think we forgot that integers support real and imag too, to match python

I think we're safe here actually, as `_real_generic` is included in the definition of `integer`:

https://github.com/numpy/numpy/blob/32b3f82e4862d85c5aebba2b2b82a931038cd103/numpy/__init__.pyi#L407

Still, it would in my opinion be cleaner to just move `real` and `imag` to the `np.generic` superclass.
As both properties have to be redefined for `np.complexfloating` anyway (due to their distinct return type) 
the original concern should not be an issue (https://github.com/numpy/numpy-stubs/pull/14#discussion_r175656159).

Reply:
I think number makes more sense than generic, as the result of str_.real is nonsense

Reply:
That might for the better, yes.
I had to double check but `real` and `imag` will apparently be deprecated for non-numeric types anyway (https://github.com/numpy/numpy/issues/10818).

In any case, I'm planning to save this for a future maintenance pull request as there are also a few other issues I'd like to fix in one go (_e.g._ `str` and `bytes` are missing from a couple of `__init__()`s).